Add Peppermint@1.4.0 w/ git auto-update

clairetsai818 opened the pull request:
Pull request for issue: #10599
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):https://github.com/wilddeer/Peppermint
 * Official website (optional, not the repository):http://wd.dizaina.net/en/scripts/peppermint/
 * NPM package url (optional):
 * License and its reference:[MIT](https://github.com/wilddeer/Peppermint/blob/master/README.md##License)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers:11
   - Count of stars:148
   - Count of forks:20


# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

clairetsai818 commented:
@cdnjs/intern3 Please help me review this, thanks!

hare1039 commented on ajax/libs/Peppermint/1.4.0/peppermint.required.css:
```diff
@@ -0,0 +1,62 @@
+/* Peppermint minimal required styles */
```
Please minify the `.css` files.

clairetsai818 commented:
I added `homepage` info, added the old versions which have different file structure and minify the files, @cdnjs/intern3 Please help me review this. Thanks!

clairetsai818 commented:
@sufuf3 I got them from GitHub repo, and I revised the commit msg. Please help me review this, thanks!

sufuf3 commented on ajax/libs/Peppermint/package.json:
```diff
@@ -0,0 +1,40 @@
+    "framework-agnostic",
+    "library-agnostic"
+  ],
+  "homepage": "http://wd.dizaina.net/en/scripts/peppermint/",
```
This change should be in the first commit.
Please delete this line in the second commit, and use `git rebase -i HEAD~2` to `edit` first commit.

clairetsai818 commented:
I make the `homepage` info in the first commit, @sufuf3 please help me review this, thanks!

clairetsai818 commented:
@pvnr0082t please help me review this PR, thanks!

clairetsai818 commented:
@pvnr0082t I think the `peppermint.js` file under the pure directory is the same as `src/peppermint.js`. Is it okay not hosting the latter one?

clairetsai818 commented:
@extend1994 @pvnr0082t The author said that we don't need to host that file. I rebase this PR, please help me review this, thanks!

clairetsai818 commented:
@extend1994 I removed those files, and here is the ref:https://github.com/wilddeer/Peppermint/issues/12
Please help me review this, thanks!

clairetsai818 commented:
@extend1994 I removed it from the first PR, and I also modified the commit msg. Please help me review this, thanks!

extend1994 commented on ajax/libs/Peppermint/1.4.0/pure/peppermint.pure.js:
```diff
@@ -0,0 +1,493 @@
+/*!
+ * Peppermint touch slider
+ * v. 1.4.0 | https://github.com/wilddeer/Peppermint
```
Hmm... But it is still here.

extend1994 commented on ajax/libs/Peppermint/1.4.0/pure/peppermint.pure.min.js:
```diff
@@ -6,4 +6,488 @@
  */
-function Peppermint(a,b){function c(a,b){for(var c in b)b.hasOwnProperty(c)&&(a[c]=b[c])}function d(a){var b=["Webkit","Moz","O","ms"],c=document.createElement("div");if(void 0!==c.style[a])return!0;a=a.charAt(0).toUpperCase()+a.slice(1);for(var d in b)if(void 0!==c.style[b[d]+a])return!0;return!1}function e(a,b){new RegExp("(\\s|^)"+b+"(\\s|$)").test(a.className)||(a.className+=" "+b)}function f ...
\ No newline at end of file
+function Peppermint(_this, options) {
```
And here too.

clairetsai818 commented:
@extend1994 Sorry that's my mistake. I remove that file in these two commits, Please help me review this.

clairetsai818 commented:
@extend1994 I modified commit msg, please help me review this, thanks!

pvnr0082t commented:
@clairetsai818 Please notify other reviewers after done some changes. :+1:

clairetsai818 commented:
@sufuf3 I edited the commit msg, please help me review this PR, thanks!
@pvnr0082t Thanks for reminding me, sorry I forgot that.

sufuf3 commented:
@clairetsai818 
The second commit message just needs to cc once.
Please modify the commit message, thank you.
```
Manually add Peppermint v1.0.0~1.2.0, cc #10599
Add Peppermint@1.0.0~1.2.0 due to different file structure
```

sufuf3 commented on ajax/libs/Peppermint/package.json:
```diff
@@ -0,0 +1,40 @@
+    "name": "Oleg Korsunsky",
+    "url": "http://wd.dizaina.net/"
+  },
+  "homepage": "http://wd.dizaina.net/en/scripts/peppermint/",
```
please use `https`

sufuf3 commented on ajax/libs/Peppermint/package.json:
```diff
@@ -0,0 +1,40 @@
+  },
+  "author": {
+    "name": "Oleg Korsunsky",
+    "url": "http://wd.dizaina.net/"
```
please use `https`
